### ISSUE TITLE
Forest generation is extreme, Causes tick lag when generating

### ISSUE THREAD

**Issue by Biochao** [Team-RTG/Realistic-Terrain-Generation#987]
- Forge version: 12.18.1.2088
- RTG version: 1.10.2-4.1.1.2

Any type of forest biome (Forest, Birch Forest, Jungle) has extremely dense tree population. During generation this slows the game to a crawl. Unless I'm missing it, I don't see a way in the config to control the tree population.

![2016-10-04_23 02 36](https://cloud.githubusercontent.com/assets/6515654/19100145/3293d5c6-8a8a-11e6-9255-dfac64db7244.png)
![2016-10-04_23 02 48](https://cloud.githubusercontent.com/assets/6515654/19100146/329573c2-8a8a-11e6-8526-754c08e4726c.png)
![2016-10-04_23 03 01](https://cloud.githubusercontent.com/assets/6515654/19100144/3293a380-8a8a-11e6-9092-b7b701b0d876.png)
![2016-10-04_23 04 06](https://cloud.githubusercontent.com/assets/6515654/19100147/3297f390-8a8a-11e6-844d-60043fa22fd5.png)

**Comment by srs-bsns** [comment]
We've already addressed a similar tick lag issue with Roofed Forest, but haven't seen it outside of that one biome. That issue is not an RTG issue, but a Forge or vanilla issue that is out of our hands.

Judging by your screenshots, the birch trees look quite dense (possibly too dense for normal RTG decoration). Can you please provide your `fml-client-latest.log`.

**Comment by Biochao** [comment]
This is the log after creating a new world and spawning right next to a dense Taiga.
[fml-client-latest.txt](https://github.com/Team-RTG/Realistic-Terrain-Generation/files/512563/fml-client-latest.txt)

![2016-10-05_20 12 37](https://cloud.githubusercontent.com/assets/6515654/19136180/64125e8e-8b38-11e6-9da2-baa436b64a48.png)
![2016-10-05_20 13 21](https://cloud.githubusercontent.com/assets/6515654/19136181/641598b0-8b38-11e6-980b-2677a102880b.png)

**Comment by whichonespink44** [comment]
@Biochao what mod is adding those trees in the screenshots? because they're not RTG trees (most of them anyways)

**Comment by whichonespink44** [comment]
I'm guessing Recurrent Complex:

```
[19:54:44] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'CorruptedEnderTree'
[19:54:44] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'WhiteWillowLarge'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SpruceTreeBig'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SpruceTree3'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SpruceTree2'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SpruceTree1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SpruceTree'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SavannaTree1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'SavannaTree'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'QuakingAspen'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'OakTreeHuge'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'OakTreeBig'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'OakTree'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'Mopane'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'LodgepolePine'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'JungleTreeHuge1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'JungleTreeHuge'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTreeSmall'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTreeHuge'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree6'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree5'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree4'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree3'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree2'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'GenericTree'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'FirTreeBig'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'FirTree'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'Fiberpalm'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'DeadTreeLarge3'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'DeadTreeLarge2'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'DeadTreeLarge1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'DeadTreeLarge'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'BirchTreeBig1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'BirchTreeBig'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'BirchTree1'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'BirchTree'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'Basswood'
[19:54:45] [Client thread/INFO] [reccomplex/reccomplex]: Registered structure 'AlascaCedar'
```

Not too familiar with RC, but... looks like the spawn frequency for those trees is way too high... maybe try decreasing them to see if that helps.

**Comment by Biochao** [comment]
Yeah, It looks like Recurrent Complex goes crazy with it's generation on a RTG world. I'll report it to Ivorius.

**Comment by srs-bsns** [comment]
The only remedy I've found for this situation is to lower your view distance when in the vacinity of dense trees.

I'm not certain of the cause, but it appears to be a possible light level update with leaves. I've not experienced this in a pure vanilla client/server setting, but I have with only Forge loaded with no mods, so it appears to be a Forge issue. We're continuing to investigate this issue.

**Comment by whichonespink44** [comment]
Potentially fixed in https://github.com/Team-RTG/Realistic-Terrain-Generation/pull/1011

Waiting on mod author to confirm before merging.

**Comment by whichonespink44** [comment]
Fixed in 6c03e26 - requires Recurrent Complex v1.2.3 or later.

**Comment by srs-bsns** [comment]
Maybe we should add a soft dependency for ReC.
